### PR TITLE
[NFC] Removed WORKAROUND_SWDEV_227826 macro and MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS envvar

### PR DESCRIPTION
**Pull request by atamazov**
Cleanup. WORKAROUND_SWDEV_227826 was disabled in #1595 > 1.5 years ago.

----
[Attribution] @junliume @JehandadKhan
- https://github.com/ROCm/MIOpen/labels/quality
- https://github.com/ROCm/MIOpen/labels/urgency_low

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1030,7 +1030,6 @@ add_custom_test(test_regression_half_vega SKIP_UNLESS_ALL FLOAT_DISABLED HALF_EN
 set_tests_properties(test_regression_half_vega PROPERTIES RUN_SERIAL On)
 
 set(ENVS_REGRESSION_ISSUE_1012
-    MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS=1
```
Comment (CAHEK7) [inline test/CMakeLists.txt]: There is one more test with this option at the line 1056.

Comment (atamazov) [reply test/CMakeLists.txt]: oops

Comment (atamazov) [reply test/CMakeLists.txt]: resolved

**Comment by xinlipn** [comment]
Hi @atamazov , Is MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS  still needed for 
test_regression_half_mi200 ?

https://github.com/ROCm/MIOpen/blob/20dcde0ed5a5f266aadcf1a75e4c460e33f326df/test/CMakeLists.txt#L1055

**Comment by CAHEK7** [comment]
> Hi @atamazov , Is MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS still needed for test_regression_half_mi200 ?
> 
> https://github.com/ROCm/MIOpen/blob/20dcde0ed5a5f266aadcf1a75e4c460e33f326df/test/CMakeLists.txt#L1055

It was deleted in the latest commit, so we don't need `MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS` at all.

**Comment by atamazov** [comment]
https://github.com/ROCm/MIOpen/labels/TESTING_CI_PASSED a while ago. I am going to update from develop now.

**Comment by atamazov** [comment]
@junliume @JehandadKhan Ready for the merging or for the next review/CI testing round.

**Comment by junliume** [comment]
Last CI round started

**Review by junliume — APPROVED** [review]

**Pull request merged**